### PR TITLE
Renumber TEE Host FIDs to match SBI draft

### PR DESCRIPTION
**Pull request by atulkharerivos**
This aligns the FIDs with the draft in [1], and has no other functional changes.

[1]:
https://github.com/riscv-non-isa/riscv-ap-tee/pull/5/commits

### CODE REVIEW

**Comment by atulkharerivos** [comment]
Related to https://github.com/rivosinc/salus/issues/112.

**Comment by abrestic-rivos** [comment]
Maybe I missed this discussion, but IMO it makes more sense to group functions by related operation/procedure than putting them in lifecycle order. For example, I would think the convert/reclaim/fence functions would belong together, as would the add measured/zero/shared pages.

**Comment by atulkharerivos** [comment]
> Maybe I missed this discussion, but IMO it makes more sense to group functions by related operation/procedure than putting them in lifecycle order. For example, I would think the convert/reclaim/fence functions would belong together, as would the add measured/zero/shared pages.

I don't have strong feelings about arranging by lifetime vs. function. In the SBI document, it lifecycle looked a little more natural (just to avoid forward references), but since it's searchable, don't see that as sticking point. Will change once others chime in.

**Comment by atishp04** [comment]
Yeah. That was the intention. I just suggested life cycle order for better reading experience. We can certainly group function based on operations as well. How about this order ? Don't feel very strongly about this but it will probably avoid bunch of forward references which was the original intention.

0. sbi_tee_host_get_tsm_info
1. sbi_tee_host_convert_pages
2. sbi_tee_host_reclaim_pages
3. sbi_tee_host_global_fence
4. sbi_tee_host_local_fence
5. sbi_tee_host_create_tvm
6. sbi_tee_host_finalize_tvm
7. sbi_tee_host_destroy_tvm
8. sbi_tee_host_add_tvm_memory_region
9. sbi_tee_host_add_tvm_page_table_pages
10. sbi_tee_host_add_tvm_measured_pages
11. sbi_tee_host_add_tvm_zero_pages
12. sbi_tee_host_add_tvm_shared_pages
13. sbi_tee_host_get_tvm_vcpu_num_register_sets
14. sbi_tee_host_get_tvm_vcpu_register_set
15. sbi_tee_host_create_tvm_vcpu
16. sbi_tee_host_run_tvm_vcpu

**Comment by abrestic-rivos** [comment]
> Yeah. That was the intention. I just suggested life cycle order for better reading experience. We can certainly group function based on operations as well. How about this order ? Don't feel very strongly about this but it will probably avoid bunch of forward references which was the original intention.
> 
> 0. sbi_tee_host_get_tsm_info
> 1. sbi_tee_host_convert_pages
> 2. sbi_tee_host_reclaim_pages
> 3. sbi_tee_host_global_fence
> 4. sbi_tee_host_local_fence
> 5. sbi_tee_host_create_tvm
> 6. sbi_tee_host_finalize_tvm
> 7. sbi_tee_host_destroy_tvm
> 8. sbi_tee_host_add_tvm_memory_region
> 9. sbi_tee_host_add_tvm_page_table_pages
> 10. sbi_tee_host_add_tvm_measured_pages
> 11. sbi_tee_host_add_tvm_zero_pages
> 12. sbi_tee_host_add_tvm_shared_pages
> 13. sbi_tee_host_get_tvm_vcpu_num_register_sets
> 14. sbi_tee_host_get_tvm_vcpu_register_set
> 15. sbi_tee_host_create_tvm_vcpu
> 16. sbi_tee_host_run_tvm_vcpu

LGTM

**Comment by atulkharerivos** [comment]
Updated per the above.

File: sbi/src/tee_host.rs [hunk 1]
```diff
@@ -256,28 +256,103 @@ impl TeeMemoryRegion {
...
         len: u64,
     },
+    /// Moves a VM from the initializing state to the Runnable state
+    ///
+    /// a6 = 6
+    Finalize {
+        /// a0 = guest id
+        guest_id: u64,
+    },
+
```
Comment (abrestic-rivos) [inline sbi/src/tee_host.rs:326]: nit: spurious newline?

**Review by abrestic-rivos — APPROVED** [review]
LGTM, thanks!

**Pull request merged**